File: frontend/src/types/ai-model.ts
// ── AI 供应商 ──────────────────────────────────────────────────────────────

export type ProviderType =
  | 'openai'
  | 'anthropic'
  | 'azure'
  | 'zhipu'
  | 'baichuan'
  | 'ollama'
  | 'custom'

export interface AIProvider {
  id: number
  tenant_id: number
  name: string
  provider_type: ProviderType
  api_base_url: string | null
  has_api_key: boolean
  config: Record<string, unknown> | null
  status: boolean
  created_at: string | null
  updated_at: string | null
}

export interface AIProviderCreateRequest {
  name: string
  provider_type: ProviderType
  api_base_url?: string
  api_key?: string
  config?: Record<string, unknown>
  status?: boolean
}

export interface AIProviderUpdateRequest {
  name?: string
  provider_type?: ProviderType
  api_base_url?: string
  api_key?: string
  config?: Record<string, unknown>
  status?: boolean
}

export interface ConnectivityTestRequest {
  model_name: string
}

export interface ConnectivityTestResult {
  success: boolean
  latency_ms: number | null
  error: string | null
}

// ── AI 模型 ───────────────────────────────────────────────────────────────

export type ModelType = 'chat' | 'embedding' | 'image' | 'audio' | 'rerank'

export interface AIModel {
  id: number
  tenant_id: number | null
  provider_id: number
  name: string
  display_name: string
  model_type: ModelType
  config: Record<string, unknown> | null
  unit_price_input: string | null
  unit_price_output: string | null
  max_context_tokens: number | null
  max_output_tokens: number | null
  status: boolean
  created_at: string | null
  updated_at: string | null
}

export interface AIModelCreateRequest {
  provider_id: number
  name: string
  display_name: string
  model_type: ModelType
  config?: Record<string, unknown>
  unit_price_input?: string
  unit_price_output?: string
  max_context_tokens?: number
  max_output_tokens?: number
  status?: boolean
}

export interface AIModelUpdateRequest {
  name?: string
  display_name?: string
  model_type?: ModelType
  config?: Record<string, unknown>
  unit_price_input?: string
  unit_price_output?: string
  max_context_tokens?: number
  max_output_tokens?: number
  status?: boolean
}

export interface ModelTestRequest {
  messages: Array<{ role: 'user' | 'assistant' | 'system'; content: string }>
}

export interface ModelTestResult {
  content: string
  prompt_tokens: number
  completion_tokens: number
  latency_ms: number
}

File: backend/app/schemas/ai_model.py
from datetime import datetime
from decimal import Decimal
from typing import Optional
from pydantic import BaseModel, Field


class AIModelCreate(BaseModel):
    provider_id: int
    name: str = Field(..., max_length=255, description="模型标识，如 gpt-4o")
    display_name: str = Field(..., max_length=255)
    model_type: str = Field(..., max_length=50, description="chat/embedding/image/audio/rerank")
    config: Optional[dict] = None
    unit_price_input: Optional[Decimal] = None
    unit_price_output: Optional[Decimal] = None
    max_context_tokens: Optional[int] = None
    max_output_tokens: Optional[int] = None
    status: bool = True


class AIModelUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=255)
    display_name: Optional[str] = Field(None, max_length=255)
    model_type: Optional[str] = Field(None, max_length=50)
    config: Optional[dict] = None
    unit_price_input: Optional[Decimal] = None
    unit_price_output: Optional[Decimal] = None
    max_context_tokens: Optional[int] = None
    max_output_tokens: Optional[int] = None
    status: Optional[bool] = None


class AIModelOut(BaseModel):
    id: int
    tenant_id: Optional[int]
    provider_id: int
    name: str
    display_name: str
    model_type: str
    config: Optional[dict]
    unit_price_input: Optional[Decimal]
    unit_price_output: Optional[Decimal]
    max_context_tokens: Optional[int]
    max_output_tokens: Optional[int]
    status: bool
    created_at: Optional[datetime]
    updated_at: Optional[datetime]

    model_config = {"from_attributes": True}


class ModelTestRequest(BaseModel):
    messages: list[dict] = Field(
        ...,
        description="消息列表，格式: [{role: 'user', content: '...'}]"
    )


class ModelTestResult(BaseModel):
    content: str
    prompt_tokens: int
    completion_tokens: int
    latency_ms: int
